tools/automate_bindings.py

```diff
@@ -0,0 +1,215 @@
+    # Get access to the global namespace
+    global_namespace = declarations.get_global_namespace(decls)
+    return global_namespace
+
+def getMethods(global_namespace):
+    Wt = global_namespace.namespace("Wt")
+    methods = Wt.member_functions()
+    main_class = (Wt.classes())[0]
+    bases = main_class.bases
+    return methods, bases
+
+def writeSourceToFile(module_name, source):
+    with open('src/luawt/%s.cpp' % module_name, 'wt') as f:
+        f.write(source)
+
+def getModuleName(filename):
+    return os.path.basename(filename)
+
+def generateIncludes(module_name):
+    return INCLUDES_TEMPLATE.lstrip() % module_name
+
+def getSelf(module_name):
+    frame = r'''
+    %s* self = luawt_checkFromLua<%s>(L, 1);
+    '''
+    return frame % (module_name, module_name)
+
+def getInbuiltTypeArgument(options):
+    frame = r'''
+    %(argument_type)s %(argument_name)s = %(func)s(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def getComplexArgument(options):
+    frame = r'''
+    %(argument_type)s* %(argument_name)s =
+        luawt_checkFromLua<%(argument_type)s>(L, %(index)s);
+    '''
+    return frame.lstrip() % options
+
+def callWtFunction(return_type, args, method_name):
+    call_s = 'self->%s(' % method_name
+    args_s = ', '.join(arg.name for arg in args)
+    func_s = call_s + args_s + ');'
+    if return_type == 'void':
+        return func_s
+    else:
+        return '%s result = %s' % (return_type, func_s)
+
+def returnValue(return_type):
+    if return_type == 'void':
+        return 'return 0;'
+    else:
+        if return_type:
+            func_name = TYPE_TO_LUA_FUNCS[return_type]
+        else:
+            func_name = 'luawt_toLua'
+        return RETURN_CALLS_TEMPLATE % func_name
+
+def getInbuiltType(type):
+    for inbuilt_type in TYPE_FROM_LUA_FUNCS:
+        if type.find(inbuilt_type) is not -1:
+            return inbuilt_type
+    return None
+
+def implementLuaCFunction(module_name, method_name, args, return_type):
+    head = 'int luawt_%s_%s(lua_State* L) {' % (module_name, method_name)
+    body = getSelf(module_name)
+    arg_n = 0
+    while arg_n < len(args):
+        arg = args[arg_n]
+        options = {
+            'argument_name' : arg.name,
+            'argument_type' : str(arg.decl_type),
+            # In Lua indices start with 1; the first one is
+            # object itself, so we have to add 2 to arg_n.
+            'index' : arg_n + 2,
+        }
+        arg_type = getInbuiltType(str(arg.decl_type))
+        if (arg_type):
+            options['func'] = TYPE_FROM_LUA_FUNCS[arg_type]
+            body += getInbuiltTypeArgument(options)
+        else:
+            body += getComplexArgument(options)
+        arg_n = arg_n + 1
+    body += callWtFunction(return_type, args, method_name)
+    return_type = getInbuiltType(str(return_type))
+    body += returnValue(return_type)
+    close = '}\n\n'
+    return head + body + close
```

Comment:
Remove `head` and `close` and put everything in one format string:
```python
return r'''int luawt_%(module)s_%(method)s(lua_State* L) {
  %(body)s
}
''' % {
  'module': module_name,
  'method': method_name,
  'body': body,
}
```

And do the same in `generateMethodsArray`.